projects/admin/src/app/record/custom-editor/holding-editor/holding-editor.component.ts

```diff
@@ -43,34 +45,59 @@ export class HoldingEditorComponent {
   /** Current error message from the backend during the serial preview example if exists */
   serialPreviewError = null;
 
+  /**  */
+  predictionModel$ = new BehaviorSubject({} as any);
+
+
+  /** Observable subscription */
+  private _subscription = new Subscription();
+
   /**
    * Constructor.
    * @param _editorService - the local editor service
    */
   constructor(private _editorService: EditorService) { }
 
+  /** Component initialization. */
+  ngOnInit() {
+    this._subscription = this.predictionModel$.pipe(
+      // wait .5s before the last change
+      debounceTime(500),
+      // only if the patterns changed
+      distinctUntilChanged((a, b) => JSON.stringify(a.patterns) === JSON.stringify(b.patterns)),
+      // cancel previous pending requests
+      switchMap(modelValue => this._editorService.getHoldingPatternPreview(
```

Comment:
`switchMap` seems great !

Reply:
Exact! I put some comment on each, because I'm pretty sure that I will forget the meaning of each in few days...